src/components/safe-apps/SafeAppActionButtons/index.tsx

```diff
@@ -0,0 +1,106 @@
+import { SAFE_APPS_EVENTS, trackSafeAppEvent } from '@/services/analytics'
+import { useCurrentChain } from '@/hooks/useChains'
+import CopyButton from '@/components/common/CopyButton'
+import ShareIcon from '@/public/images/common/share.svg'
+import BookmarkIcon from '@/public/images/apps/bookmark.svg'
+import BookmarkedIcon from '@/public/images/apps/bookmarked.svg'
+import DeleteIcon from '@/public/images/common/delete.svg'
+
+type SafeAppActionButtonsProps = {
+  safeApp: SafeAppData
+  isBookmarked?: boolean
+  onBookmarkSafeApp?: (safeAppId: number) => void
+  removeCustomApp?: (safeApp: SafeAppData) => void
+}
+
+const SafeAppActionButtons = ({
+  safeApp,
+  isBookmarked,
+  onBookmarkSafeApp,
+  removeCustomApp,
+}: SafeAppActionButtonsProps) => {
+  const router = useRouter()
+  const currentChain = useCurrentChain()
+
+  const isCustomApp = safeApp.id < 1
+  const shareSafeAppUrl = getShareSafeAppUrl(router, safeApp.url, currentChain)
+
+  const handleCopyShareSafeAppUrl = () => {
+    const appName = isCustomApp ? safeApp.url : safeApp.name
+    trackSafeAppEvent(SAFE_APPS_EVENTS.COPY_SHARE_URL, appName)
+  }
+
+  return (
+    <Box display="flex" gap={1}>
+      {/* Copy share Safe App url button */}
+      <CopyButton
+        initialToolTipText={`Copy share URL for ${safeApp.name}`}
+        onCopy={handleCopyShareSafeAppUrl}
+        text={shareSafeAppUrl}
+      >
+        <SvgIcon component={ShareIcon} inheritViewBox color="border" fontSize="small" />
+      </CopyButton>
+
+      {/* Bookmark Safe App button */}
+      {onBookmarkSafeApp && (
+        <Tooltip title={`${isBookmarked ? 'Unpin' : 'Pin'} ${safeApp.name}`} placement="top">
+          <IconButton
+            size="small"
+            onClick={(event) => {
+              event.preventDefault()
+              event.stopPropagation()
+              onBookmarkSafeApp(safeApp.id)
+            }}
+          >
+            <SvgIcon
+              component={isBookmarked ? BookmarkedIcon : BookmarkIcon}
+              inheritViewBox
+              color={isBookmarked ? 'primary' : undefined}
+              fontSize="small"
+            />
+          </IconButton>
+        </Tooltip>
+      )}
+
+      {/* Remove Custom Safe App button */}
+      {removeCustomApp && (
+        <Tooltip title={`Delete ${safeApp.name}`} placement="top">
+          <IconButton
+            size="small"
+            onClick={(event) => {
+              event.preventDefault()
+              event.stopPropagation()
+              removeCustomApp(safeApp)
+            }}
+          >
+            <SvgIcon component={DeleteIcon} inheritViewBox fontSize="small" />
+          </IconButton>
+        </Tooltip>
+      )}
+    </Box>
+  )
+}
+
+export default SafeAppActionButtons
+
+export const getShareSafeAppUrl = (router: NextRouter, appUrl: string, currentChain?: ChainInfo) => {
+  const shareUrlObj: UrlObject = {
+    protocol: typeof window !== 'undefined' ? window.location.protocol : '',
```

Comment:
This might cause hydration issues because you're accessing the window object outside of an effect.

Reply:
I am just using the same logic that [its defined in our `AppCard.tsx` component in dev](https://github.com/safe-global/web-core/blob/dev/src/components/safe-apps/AppCard/index.tsx#L194) (I just created a `getShareSafeAppUrl` function to abstract it). Do you think it's worth to update it?

Reply:
Ah, let's keep it then, I'll make a tech debt issue.